src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -92,6 +92,7 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
         # create the RGW keyring
         rgw_user = f'{rados_user}-rgw'
         rgw_keyring = self.create_rgw_keyring(daemon_spec)
+        bind_addr = spec.virtual_ip if spec.virtual_ip else (daemon_spec.ip if daemon_spec.ip else '')
```

Comment:
should we error in case the ip is empty?

Reply:
I was just maintaining the previous behavior here, which put an empty string if `daemon_spec.ip` was `None`. We _probably_ could error if it's `None` but I don't know exactly why it was set to default to an empty string in the first place so I'm not sure I want to touch it. Perhaps it was just a typing thing to make sure the value is always a string.

Reply:
That's right. I'd also recomend not altering the previous behaviour. In order to do not alter the legacy logic maybe just some `warning` log to notify the user that bind_addr is left empty. That could help in some weired situation.

Reply:
makes sense. I'll add the warning log at least.

Reply:
added warning level log message for if this actually is getting set to an empty string